File: src/main/java/de/hinundhergestellt/jhuh/service/ready2order/GroupArtooMappedProduct.java
package de.hinundhergestellt.jhuh.service.ready2order;

import de.hinundhergestellt.jhuh.vendors.ready2order.ArtooProductGroup;

import java.util.List;

public class GroupArtooMappedProduct extends ArtooMappedProduct {

    private final ArtooProductGroup group;

    GroupArtooMappedProduct(ArtooProductGroup group, List<ArtooMappedVariation> variations) {
        super(variations);
        this.group = group;
    }

    @Override
    public String getName() {
        return group.getName();
    }
}

File: src/main/java/de/hinundhergestellt/jhuh/service/ready2order/GroupArtooMappedProduct.kt
package de.hinundhergestellt.jhuh.service.ready2order

import de.hinundhergestellt.jhuh.vendors.ready2order.ArtooProductGroup

class GroupArtooMappedProduct internal constructor(
    private val group: ArtooProductGroup,
    variations: List<ArtooMappedVariation>
) : ArtooMappedProduct(variations) {

    override val id = "group-${group.id}"
    override val name by group::name
    override val description by group::description
}
